include/gigamonkey/p2p/AbstractAddressManager.hpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2022 Katrina Knight
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef GIGAMONKEY_P2P_ABSTRACTADDRESSMANAGER_HPP_
+#define GIGAMONKEY_P2P_ABSTRACTADDRESSMANAGER_HPP_
+
+#include <string>
+#include <list>
+#include <boost/heap/priority_queue.hpp>
+#include "data/encoding/endian/arithmetic.hpp"
+#include "address.hpp"
+namespace Gigamonkey::Bitcoin::P2P {
+	struct NodeAddress {
+		std::string IpAddress{};
+		data::uint16_big port{};
+		data::uint64_little services{};
+		data::int32_little timestamp{};
```

Comment:
Why would addresses include services and timestamp? That's not an address. Also this supports ipv6 right? We want to do everything on ipv6 in Bitcoin eventually.